libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs

```diff
@@ -42,30 +118,137 @@ impl Connector for PostgresDatamodelConnector {
         &self.constructors
     }
 
-    fn parse_native_type(&self, name: &str, args: Vec<u32>) -> Option<NativeTypeInstance> {
+    fn parse_native_type(
+        &self,
+        name: &str,
+        args: Vec<u32>,
+        scalar_type: ScalarType,
+    ) -> Result<NativeTypeInstance, ConnectorError> {
         let constructor = self.find_native_type_constructor(name);
+        let length = *args.len();
         let native_type = match name {
+            SMALL_INT_TYPE_NAME => PostgresType::SmallInt,
+            INTEGER_TYPE_NAME => PostgresType::Integer,
             BIG_INT_TYPE_NAME => PostgresType::BigInt,
+            REAL_TYPE_NAME => PostgresType::Real,
+            DOUBLE_PRECISION_TYPE_NAME => PostgresType::DoublePrecision,
+            SMALL_SERIAL_TYPE_NAME => PostgresType::SmallSerial,
+            SERIAL_TYPE_NAME => PostgresType::Serial,
+            BIG_SERIAL_TYPE_NAME => PostgresType::BigSerial,
             VARCHAR_TYPE_NAME => {
-                let length = *args.first().unwrap();
-                PostgresType::VarChar(length)
+                if let Some(arg) = *args.first() {
+                    PostgresType::VarChar(arg)
+                }
+            }
+            CHAR_TYPE_NAME => {
+                if let Some(arg) = *args.first() {
+                    PostgresType::Char(arg)
+                }
+            }
+            TEXT_TYPE_NAME => PostgresType::Text,
+            TIMESTAMP_TYPE_NAME => {
+                if let Some(arg) = *args.first() {
+                    PostgresType::Timestamp(arg as u8)
+                }
+            }
+            TIMESTAMP_WITH_TIMEZONE_TYPE_NAME => {
+                if let Some(arg) = *args.first() {
+                    PostgresType::TimestampWithTimeZone(arg as u8)
+                }
+            }
```

Comment:
Again, I reckon this has to be added, but don't know how to express it 😅 

```suggestion
            TIMESTAMP_WITH_TIMEZONE_TYPE_NAME => {
                if let Some(arg) = *args.first() {
                    PostgresType::TimestampWithTimeZone(arg as u8)
                }
            }
            TIMESTAMP_WITH_TIMEZONE_RANGE_TYPE_NAME => {
                if let Some(arg) = *args.first() {
                    PostgresType::TimestampWithTimeRangeZone(arg as u8)
                }
            }
```